### PR TITLE
XIVY-14730 disable metadata of overwritten variables

### PR DESCRIPTION
**Pull request by ivy-lgi**
Make the metadata read only if the variable is found in the known variables of the required projects.

### CODE REVIEW

File: packages/variable-editor/src/components/variables/detail/DetailContent.tsx [hunk 1]
```diff
@@ -1,16 +1,33 @@
...
 import { Metadata } from './Metadata';
 import { Value } from './Value';
-import './DetailContent.css';
-import { useMemo } from 'react';
+
+export const useOverwrites = () => {
+  const { context, variables, selectedVariable } = useAppContext();
+  const variableNodes = getNodesOnPath(variables, selectedVariable);
+  let currentNode: ProjectVarNode | undefined = useMeta('meta/knownVariables', context, EMPTY_PROJECT_VAR_NODE).data;
+  for (const variableNode of variableNodes) {
```
[the remark this answers is not in the document]
Comment (ivy-lgi) [reply packages/variable-editor/src/components/variables/detail/DetailContent.tsx:16]: I don't think this is an improvement.
It saves one call to `currentNode.children.find` for a call to `currentNode.children.length === 0` on every iteration.
Or do you mean something else than the following?

```typescript
  for (const variableNode of variableNodes) {
    if (currentNode.children.length === 0) {
      return false;
    }
    currentNode = currentNode.children.find(child => child.name === variableNode?.name);
    if (!currentNode) {
      return false;
    }
  }
  return true;
```

[the remark this answers is not in the document]
Comment (ivy-lli) [reply packages/variable-editor/src/components/variables/detail/DetailContent.tsx:16]: nope more like
```suggestion
  let currentNode: ProjectVarNode | undefined = useMeta('meta/knownVariables', context, EMPTY_PROJECT_VAR_NODE).data;
  if (currentNode === undefined || currentNode.children.length === 0) {
    return false;
  }
  const variableNodes = getNodesOnPath(variables, selectedVariable);
  for (const variableNode of variableNodes) {
```

**Review by ivy-lli — APPROVED** [review]
I would consider the fast return as improvement, but otherwise good to go 👍

File: packages/variable-editor/src/components/variables/detail/DetailContent.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ivy-lgi) [reply packages/variable-editor/src/components/variables/detail/DetailContent.tsx]: Ah yes, saving the call to `getNodesOnPath`. Makes sense, thanks :+1:

**Pull request merged**